test/es6/bug_issue_2747.js

```diff
@@ -0,0 +1,29 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var oldSet = Map.prototype.set;
+var m;
+function constructorFunc () {
+    m = new Map([["a", 1], ["b", 2]]);
+}
+
+Object.defineProperty(Map.prototype, "set", {
+    get: Map.prototype.get, // can be any Map.prototype method that depends on `this` being a valid map
+    configurable: true
+});
+assert.throws(function () { return Map.prototype.set });

```

Comment:
throws function takes other params, which actually describes what kind of error and description. It will be useful for readability. 
same true for doesNotThrow as well.